addons/mail/static/tests/qunit_suite_tests/models/file_uploader_tests.js

```diff
@@ -50,6 +53,70 @@ QUnit.test("no conflicts between file uploaders", async function () {
     await contains(".o_Chatter .o_AttachmentCard");
     await contains(".o_ChatWindow .o_Message .o_AttachmentCard");
 });
+
+QUnit.test('Arrows on nonviewable attachment', async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId = pyEnv['res.partner'].create({});
+    const irAttachmentId = pyEnv['ir.attachment'].create({
+        mimetype: 'text/plain',
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+    pyEnv['mail.message'].create({
+        attachment_ids: [irAttachmentId],
+        model: 'res.partner',
+        res_id: resPartnerId,
+    });
+    pyEnv['res.partner'].write([resPartnerId], {message_main_attachment_id : irAttachmentId})
+    const views = {
+        'res.partner,false,form':
+            '<form string="Partners">' +
+                '<sheet>' +
+                    '<field name="name"/>' +
+                '</sheet>' +
+                '<div class="o_attachment_preview"/>' +
+                '<div class="oe_chatter">' +
+                    '<field name="message_ids"/>' +
+                '</div>' +
+            '</form>',
+    };
+    patchUiSize({ size: SIZES.XXL });
+    const { messaging, openFormView } = await start({
+        mockRPC(route, args) {
+            if (_.str.contains(route, '/web/static/lib/pdfjs/web/viewer.html')) {
+                var canvas = document.createElement('canvas');
+                return canvas.toDataURL();
+            }
+        },
+        serverData: { views },
+    });
+    await openFormView({
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+
+    // No viewer as text file is non viewable
+    await contains(".oe_chatter");
+    await contains(".o_attachment_preview", { count: 0 });
+    await contains(".o_attachment_preview_container > iframe", { count: 0 });
+    await contains(".arrow", { count: 0 });
```

Comment:
I would remove these assertions completely, they are not useful for the interactions below. What matters is the assertion after interaction.